### ISSUE TITLE
Optimal Temp and OOD average score

### ISSUE THREAD

**Issue by mostafaelaraby** [hongxin001/logitnorm_ood#4]
I have some questions : 

- Why is optimal T computed on the test set? [this line](https://github.com/hongxin001/logitnorm_ood/blob/0a60eeffb7dfc970fe68e07c5649ea1c9c8244c6/test.py#L88)
- Do we have two different temperatures, one for logitnorm loss and one for MSP computation? (as it seems so from the code)
- Why do we compute the results on the average of10 shuffled subsets of the OOD data having a fifth of the total OOD dataset .as shown in [this line](https://github.com/hongxin001/logitnorm_ood/blob/0a60eeffb7dfc970fe68e07c5649ea1c9c8244c6/test.py#L95), wouldn't it be better to compute it one time on the whole OOD dataset?

Thanks for the nice code

**Comment by hongxin001** [comment]
Thanks for your attention. 

1. This optimal T is for Temperature scaling in confidence calibration. We do not need this for OOD detection.
2. No. The T shown in the test.py is for confidence calibration. We need the T for OOD detection only in the case of energy score.
3. This is a common operation in the previous papers, like energy score, ODIN. So we follow this way here.

**Comment by mostafaelaraby** [comment]
One final question in test.py [Argument out_as_pos](https://github.com/hongxin001/logitnorm_ood/blob/0a60eeffb7dfc970fe68e07c5649ea1c9c8244c6/test.py#L148) 
Does this mean we are measuring how well we fit the training data rather than its performance to identify OOD samples?

**Comment by hongxin001** [comment]
Nope. This argument is to set positive as OOD examples or ID examples. You may refer to the code of energy score ( https://github.com/wetliu/energy_ood/blob/77f3c09b788bb5a7bfde6fd3671228320ea0949c/CIFAR/test.py#L43) to understand this part.
